src/mpid/ch4/netmod/ofi/ofi_rma.h

```diff
@@ -170,6 +169,19 @@ static inline void MPIDI_OFI_query_acc_atomic_support(MPI_Datatype dt, int query
     MPIR_FUNC_VERBOSE_EXIT(MPID_STATE_MPIDI_OFI_QUERY_ACC_ATOMIC_SUPPORT);
 }
 
+MPL_STATIC_INLINE_PREFIX void MPIDI_OFI_sigreq_cntr_decr(MPIR_Request ** sigreq)
+{
+    if (sigreq) {
+        int tmp;
+        /* If sigreq is not NULL, *sigreq should be a valid object after
+         * returning from MPIDI_OFI_INIT_SIGNAL_REQUEST(). The allocation of
+         * *sigreq is inside MPIDI_OFI_INIT_SIGNAL_REQUEST() or from upper level,
+         * depending on MPIDI_CH4_MT_MODEL. */
+        MPIR_Assert(*sigreq != NULL);
+        MPIR_cc_decr((*sigreq)->cc_ptr, &tmp);
```

Comment:
Call `MPID_Request_complete` instead of modifying the cc directly.

Reply:
I think we cannot call `MPID_Request_complete` here, since it will free up object when hitting zero. We want to keep object instead.
Here, when set RMA req counter, we set it be zero but keep the req object.

Current code is doing set(cc_ptr, 0), while I change it to `create, and cc_decr` to be thread safe.

Reply:
That should be fixed with refcounting. How are references tracked for these types of requests?

Reply:
It is just normal MPID_Request created by `MPIDI_OFI_REQUEST_CREATE_CONDITIONAL`. 
cntr hits zero, ref count decrease and hit zero and free.

You are thinking about adding an extra ref count incr when creating request object in `MPIDI_OFI_INIT_SIGNAL_REQUEST`?

Reply:
Yes, it seems to be necessary. If 2 threads are checking the request, they each should hold a reference to it.

Reply:
okay, let me change and check it.